sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/Utils.java

```diff
@@ -93,18 +93,48 @@ private static ObjectMapper createAndInitializeObjectMapper(boolean allowDuplica
         }
         objectMapper.configure(DeserializationFeature.ACCEPT_FLOAT_AS_INT, false);
 
-
-        // We will not register after burner for java 16+, due to its breaking changes
-        // https://github.com/Azure/azure-sdk-for-java/issues/23005
-        if (JAVA_VERSION != -1 && JAVA_VERSION < 16) {
-            objectMapper.registerModule(new AfterburnerModule());
-        }
+        tryToLoadJacksonPerformanceLibrary(objectMapper);
 
         objectMapper.registerModule(new JavaTimeModule());
 
         return objectMapper;
     }
 
+    private static void tryToLoadJacksonPerformanceLibrary(ObjectMapper objectMapper) {
+        // Afterburner and Blackbird are libraries that increase the performance of marshaling json to objects
+        boolean loaded = false;
+        if (JAVA_VERSION != -1) {
+            if (JAVA_VERSION >= 11) {
+                // Blackbird is preferred and only works with java 11+
+                // https://github.com/FasterXML/jackson-modules-base/tree/2.18/blackbird
+                loaded = loadModuleIfFound("com.fasterxml.jackson.module.blackbird.BlackbirdModule", objectMapper);
+            }
+            if (!loaded && JAVA_VERSION < 16) {
+                // Afterburner no longer works with java 16
+                // https://github.com/Azure/azure-sdk-for-java/issues/23005
+                // https://github.com/FasterXML/jackson-modules-base/tree/2.18/afterburner
+                loaded = loadModuleIfFound("com.fasterxml.jackson.module.afterburner.AfterburnerModule", objectMapper);
+            }
+        }
+        if (!loaded) {
+            logger.warn("Neither Afterburner nor Blackbird Jackson module loaded.  Consider adding one to your classpath to for maximum Jackson performance.");
```

Comment:
```suggestion
            logger.warn("Neither Afterburner nor Blackbird Jackson module loaded.  Consider adding one to your classpath for maximum Jackson performance.");
```
Nit: I think this was a typo.

Reply:
Fixed